### PR TITLE
feat(ec2): EBS volume configuration for BastionHostLinux 

### LINKED ISSUE #6945: BastionHostLinux does not encrypt EBS volume
BastionHostLinux does not encrypt EBS Volume, and does not provide the option to encrypt the EBS volume. In version 1.30, I would have to manually enable encryption on EBS volumes.

### Use Case

Compliance regulations could require us to secure all EBS volumes.

### Proposed Solution

Allow a user to encrypt an EBS volume. To stay consistent, it'd be nice to do something similar to [DatabaseCluster](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-rds.DatabaseCluster.html)

```
kmsKey?,
storageEncrypted?
```

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by shreyasdamle**
### Commit Message
feat(ec2): EBS volume configuration for BastionHostLinux

Added `blockDevices?` property to `BastionHostLinuxProps` to allow full EBS device configuration so that customers can use it for use-cases like to encrypt an EBS volume for BastionHostLinux

Fixes #6945 
### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by shreyasdamle** [comment]
Sorry, I'm not sure why this request is showing commits that are already in `master` branch. However, this link will show the correct comparison: https://github.com/aws/aws-cdk/compare/master...shreyasdamle:sdamle-ec2?diff=split

**Comment by shreyasdamle** [comment]
@rix0rrr Thanks for the review.

I have removed unrelated changes from this PR. Please let me know if I missed anything.

File: packages/@aws-cdk/aws-ec2/lib/bastion-host.ts [hunk 1]
```diff
@@ -64,6 +65,14 @@ export interface BastionHostLinuxProps {
...
    */
   readonly machineImage?: IMachineImage;
+
+  /**
+   * Encryption for EBS volume
+   * If true, encrypted volume will be created with a default voulme size of 10 GiB.
+   *
+   * @default false
+   */
+  readonly ebsVolumeEncryption?: boolean;
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-ec2/lib/bastion-host.ts:75]: I think if we're going to go here, we need full EBS device configuration support, just like we have on `ec2.Instance`.

Comment (shreyasdamle) [reply packages/@aws-cdk/aws-ec2/lib/bastion-host.ts]: Thanks for the feedback!

I have made the requested changes.

**Pull request merged**